src/Features/Core/Portable/EmbeddedLanguages/AbstractLanguageDetector.cs

```diff
@@ -93,34 +112,138 @@ public bool IsPossiblyPatternToken(SyntaxToken token, ISyntaxFacts syntaxFacts)
+            var parent = syntaxFacts.WalkUpParentheses(token.Parent);
 
-            return IsEmbeddedLanguageString(token, argumentNode, semanticModel, cancellationToken, out options);
+            if (syntaxFacts.IsArgument(parent.Parent))
+            {
+                var argument = parent.Parent;
+                if (IsArgumentToWellKnownAPI(token, argument, semanticModel, cancellationToken, out options))
+                    return true;
+
+                if (IsArgumentToParameterWithMatchingStringSyntaxAttribute(semanticModel, argument, cancellationToken, out options))
+                    return true;
+            }
+            else
+            {
+                var statement = parent.FirstAncestorOrSelf<SyntaxNode>(syntaxFacts.IsStatement);
+                if (syntaxFacts.IsSimpleAssignmentStatement(statement))
+                {
+                    syntaxFacts.GetPartsOfAssignmentStatement(statement, out var left, out var right);
+                    if (parent == right &&
+                        IsFieldOrPropertyWithMatchingStringSyntaxAttribute(semanticModel, left, cancellationToken))
+                    {
+                        return true;
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        private bool IsArgumentToParameterWithMatchingStringSyntaxAttribute(SemanticModel semanticModel, SyntaxNode argumentNode, CancellationToken cancellationToken, out TOptions options)
+        {
+            var operation = semanticModel.GetOperation(argumentNode, cancellationToken);
+            if (operation is IArgumentOperation { Parameter: { } parameter } &&
+                HasMatchingStringSyntaxAttribute(parameter))
+            {
+                options = GetOptionsFromSiblingArgument(argumentNode, semanticModel, cancellationToken) ??
+                          GetStringSyntaxDefaultOptions();
+                return true;
+            }
+
+            options = default;
+            return false;
+        }
+
+        private bool IsFieldOrPropertyWithMatchingStringSyntaxAttribute(
+            SemanticModel semanticModel, SyntaxNode left, CancellationToken cancellationToken)
+        {
+            var symbol = semanticModel.GetSymbolInfo(left, cancellationToken).Symbol;
+            return symbol is IFieldSymbol or IPropertySymbol &&
+                HasMatchingStringSyntaxAttribute(symbol);
+        }
+
+        private bool HasMatchingStringSyntaxAttribute(ISymbol symbol)
+        {
+            foreach (var attribute in symbol.GetAttributes())
+            {
+                if (IsMatchingStringSyntaxAttribute(attribute))
+                    return true;
+            }
+
+            return false;
+        }
+
+        private bool IsMatchingStringSyntaxAttribute(AttributeData attribute)
+        {
+            if (attribute.ConstructorArguments.Length == 0)
+                return false;
+
+            if (attribute.AttributeClass is not
```

Comment:
I really wish there was a cleaner way to write this...

Reply:
indeed!  @333fred any thoughts on this?